File: 백준/Silver/1149. RGB거리/RGB거리.py
import sys
input = sys.stdin.readline

n = int(input())
rgb = [0]*3
for _ in range(n):
    r, g, b = rgb[:]
    cur_r, cur_g, cur_b = map(int, input().split())
    cur_r += min(g, b)
    cur_g += min(b, r)
    cur_b += min(r, g)
    rgb = [cur_r, cur_g, cur_b]

print(min(rgb))

File: 백준/Silver/1149. RGB거리/RGB거리.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt').toString().split('\n')

const N = Number(input[0])
const cost = []
for (let i = 1; i < N + 1; i++) {
  cost.push(input[i].split(' ').map(Number))
}

const colors = [[0, 1, 2], [1, 2, 0], [2, 0, 1]]
for (let i = 1; i < N; i++) {
  for (let [a, b, c] of colors) {
    cost[i][a] += Math.min(cost[i - 1][b], cost[i - 1][c])
  }
}

console.log(Math.min(...cost[N - 1]))